crates/agent/src/controllers/capture.rs

```diff
@@ -85,37 +65,41 @@ async fn maybe_publish<C: ControlPlane>(
             );
             Ok((detail, draft_capture))
         })
-        .await?;
-    if published {
-        return Ok(true);
+        .await;
+    if dependencies_published.as_ref().ok() == Some(&true) {
```

Comment:
nit: can this be simplified as below?
```suggestion
    if dependencies_published.as_ref() == Ok(&true) {
```

Reply:
It can't, because the anyhow error doesn't impl `PartialEq` (can't be compared). That's why I used `.ok()`

Reply:
aaaah I see, interesting!